.github/workflows/ci.yml

```diff
@@ -771,6 +771,22 @@ jobs:
           ./.github/bin/build-pt-matrix-from-impacted-connectors.py -v -m .github/test-pt-matrix.yaml -o matrix.json
       - name: Build PT matrix (impacted-features)
         if: steps.filter.outputs.product-tests == 'false' && !contains(github.event.pull_request.labels.*.name, 'tests:all') && !contains(github.event.pull_request.labels.*.name, 'product-tests:all')
+        # all these envs are required to be set by some product test environments
+        env:
+          ABFS_CONTAINER:
```

Comment:
@findinpath @alexjo2144 just FYI, it's a bit weird we have to provide so many variables just to describe the environment:
```bash
DATABRICKS_73_JDBC_URL= DATABRICKS_91_JDBC_URL= DATABRICKS_104_JDBC_URL= DATABRICKS_LOGIN= DATABRICKS_TOKEN= AWS_REGION= S3_BUCKET= DATABRICKS_AWS_ACCESS_KEY_ID= DATABRICKS_AWS_SECRET_ACCESS_KEY= testing/bin/ptl suite describe --suite suite-delta-lake-databricks --config config-default
```

but I do realize using `requireNonNull` allows for providing an actionable error message. I'm not sure how to delay getting the actual values to when the env is started, not just created, so we can keep this as is.

Reply:
If someone adds a new environment variable and forgets to add it here, what happens?

Reply:
Yes, I made sure failures like that are not silently ignored.

Reply:
`DATABRICKS_XX_JDBC_URL` corresponds to the Databricks cluster. There are 3 clusters we check for compatibility against. The suite contains correspondingly 3 environments against which we test Delta Lake.

AWS related settings are needed for testing against AWS S3 & Glue (the metastore for the Delta Lake tests).

> If someone adds a new environment variable and forgets to add it here, what happens ?

If the env variable is not related to the product tests it will simply be ignored.